docs/src/main/asciidoc/async-message-passing.adoc

```diff
@@ -0,0 +1,239 @@
+include::./attributes.adoc[]
+= {project-name} - Asynchronous messages between beans
+
+{project-name} allows different beans to interact using asynchronous messages, enforcing loose-coupling.
+The messages are sent to _virtual addresses_.
+It offers 3 types of delivery mechanism:
+
+- point-to-point - send the message, one consumer receives it. If several consumers listen to the address, a round robin is applied;
+- publish/subscribe - publish a message, all the consumers listening to the address are receiving the message;
+- request/reply - send the message and expect a response. The receiver can respond to the message in an asynchronous-fashion
+
+All these delivery mechanism are non-blocking, and are providing one of the fundamental brick to build reactive applications.
+
+== Installing
+
+This mechanism uses the Vert.x EventBus, so you need to enable the `vertx` extension to use this feature.
+If you are creating a new project, set the `extensions` parameter are follows:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=vertx-quickstart \
+    -Dextensions="vert.x"
```

Comment:
```suggestion
    -Dextensions="vertx"
```

Reply:
Both are supported, but yes, `vertx` is one less keystroke.